extensions/wikia/AdEngine/js/lookup/amazonMatch.js

```diff
@@ -145,21 +147,30 @@ define('ext.wikia.adEngine.lookup.amazonMatch', [
 		return amazonCalled;
 	}
 
+	function wasRendered() {
+		log(['wasRendered', amazonRendered], 'debug', logGroup);
+		return amazonRendered;
+	}
+
 	function getSlotParams(slotName) {
 		log(['getSlotParams', slotName], 'debug', logGroup);
 
 		var amznSlots = [];
 
-		Object.keys(sizeMapping).forEach(function (amazonSize) {
-			var validSlotNames = sizeMapping[amazonSize],
-				amazonPricePoint = bestPricePointForSize[amazonSize];
+		if (!wasRendered()) {
+			Object.keys(sizeMapping).forEach(function (amazonSize) {
+				var validSlotNames = sizeMapping[amazonSize],
+					amazonPricePoint = bestPricePointForSize[amazonSize];
 
-			if (validSlotNames.indexOf(slotName) !== -1 && amazonPricePoint) {
-				amznSlots.push('a' + amazonSize + 'p' + amazonPricePoint);
-			}
-		});
+				if (validSlotNames.indexOf(slotName) !== -1 && amazonPricePoint) {
+					amznSlots.push('a' + amazonSize + 'p' + amazonPricePoint);
+				}
+			});
 
-		log(['getSlotParams - amznSlots: ', amznSlots], 'debug', logGroup);
+			log(['getSlotParams - amznSlots: ', amznSlots], 'debug', logGroup);
+		} else {
+			log(['getSlotParams - no amznSlots since ads has been already displayed', slotName], 'debug', logGroup);
```

Comment:
I wonder if this else statement is needed since we have __wasRendered__ method with log, or maybe this method is useless because it's not used outside the module anymore?